src/Shell/Task/ActionTask.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace HavokInspiration\ActionsClass\Shell\Task;
+
+use Cake\Console\Shell;
+use Bake\Shell\Task\SimpleBakeTask;
+use Cake\Core\Configure;
+
+/**
+ * Command line to create HavokInspiration\ActionsClass action file
+ */
+class ActionTask extends SimpleBakeTask
+{
+	/**
+     * Tasks to be loaded by this Task
+     *
+     * @var array
+     */
+	public $tasks = [
+		'Bake.BakeTemplate'
+	];
+
+    /**
+     * {@inheritDoc}
+     */
+    public function name()
+    {
+        return 'file for HavokInspiration/ActionsClass';
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function fileName($name)
+    {
+        return $name . '.php';
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function template()
+    {
+        return 'HavokInspiration/ActionsClass.action';
+    }
+
+    /**
+     * Generate a class stub
+     *
+     * @param string $name The classname to generate.
+     * @return string
+     */
+    public function bake($name)
+    {
+    	$this->out("\n" . sprintf('Baking action class for %s...', $name), 1, Shell::QUIET);
+
+    	if (strpos($name, '/') !== false) {
+	    	list($controller, $action) = explode('/', $name);
+    	} else {
+    		$controller = $name;
+    		$action = 'index';
+    	}
+		$controller = $this->_camelize($controller);
+		$action = $this->_camelize($action);
+
+        $path = APP . 'Controller';
+        $namespace = Configure::read('App.namespace');
+
+    	$prefix = $this->_getPrefix();
+    	if ($prefix) {
+            $path .= DS . $prefix;
+            $prefix = '\\' . str_replace('/', '\\', $prefix);
+        }
+
+        if ($this->plugin) {
+            $path = $this->getPath(). 'Controller';
+            $namespace = $this->_pluginNamespace($this->plugin);
+        }
+
+        $data = [
+	        'action' 		=> $action,
+	        'controller' 	=> $controller,
+        	'namespace' 	=> $namespace,
+            'prefix'        => $prefix
+        ];
```

Comment:
Indentation is off (and only one space between `'` and `=>`. They should not be aligned they tend to be out of sync really quickly.